cmake/lfc.cmake

```diff
@@ -61,15 +61,16 @@ function(lf_build_generated_code MAIN_TARGET SOURCE_GEN_DIR)
   endif()
 
   # Check if the Include.cmake file exists in SOURCE_GEN_DIR
-  if (NOT EXISTS ${SOURCE_GEN_DIR}/Include.cmake)
+  if (NOT EXISTS ${SOURCE_GEN_DIR}/${FEDERATE}/Include.cmake)
```

Comment:
Instead of introducing a global variable, the caller of lf_build_generated_code should pass the correct SOURCE_GEN_DIR to this function. If you code-generate a federated program, source-gen dir must include the federate name